extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/health/MongoHealthCheck.java

```diff
@@ -10,26 +15,49 @@
 
 import com.mongodb.client.MongoClient;
 
+import io.quarkus.arc.Arc;
+
 @Readiness
 @ApplicationScoped
 public class MongoHealthCheck implements HealthCheck {
-    @Inject
-    MongoClient mongoClient;
+    private static final String DEFAULT_CLIENT = "__default__";
+    private Map<String, MongoClient> clients = new HashMap<>();
```

Comment:
Does this perhaps need to be a `ConcurrentHashMap`?

Reply:
We use the same inside Agroal: https://github.com/quarkusio/quarkus/blob/master/extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/health/DataSourceHealthCheck.java

As there is only one MongoHealthCheck bean and the `clients` attribute is not static and the `PostConstruct` should be runned once by Arc at init time I think it's safe to use a regular `HashMap` here.

Reply:
That's what I thought, I just wanted to make sure we aren't missing something here, thanks